crac-plugin/src/main/java/io/micronaut/gradle/crac/CRaCCheckpointDockerfile.java

```diff
@@ -0,0 +1,153 @@
+package io.micronaut.gradle.crac;
+
+import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
+import io.micronaut.gradle.docker.DockerBuildStrategy;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.plugins.BasePlugin;
+import org.gradle.api.provider.ListProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@CacheableTask
+public class CRaCCheckpointDockerfile extends Dockerfile {
+    public static final String DEFAULT_WORKING_DIR = "/home/app";
+
+    @Input
+    private final Property<String> baseImage;
```

Comment:
Still think this task should be converted to an abstract class and let Gradle implement the getters injection.